inc/Ui/Admin.php

```diff
@@ -0,0 +1,172 @@
+        $this->showManagerMenu();
+        $this->showVersion();
+        $this->showPluginMenu();
+        echo '</div>';
+    }
+
+    /**
+     * Display the standard admin tasks
+     */
+    protected function showAdminMenu() {
+        /** @var \DokuWiki_Auth_Plugin $auth */
+        global $auth;
+        global $INFO;
+
+        if(!$INFO['isadmin']) return;
+
+        // user manager only if the auth backend supports it
+        if(!$auth || !$auth->canDo('getUsers') ) {
+            if(isset($this->menu['usermanager'])) unset($this->menu['usermanager']);
+        }
+
+        echo '<ul class="admin_tasks">';
+        foreach(array('usermanager','acl', 'extension', 'config', 'styling') as $plugin) {
+            if(!isset($this->menu[$plugin])) continue;
+            $this->showMenuItem($this->menu[$plugin]);
+            unset($this->menu[$plugin]);
+        }
+        echo '</ul>';
+    }
+
+    /**
+     * Display the standard manager tasks
+     */
+    protected function showManagerMenu() {
+        echo '<ul class="admin_tasks">';
+        foreach(array('revert','popularity') as $plugin) {
+            if(!isset($this->menu[$plugin])) continue;
+            $this->showMenuItem($this->menu[$plugin]);
+            unset($this->menu[$plugin]);
+        }
+        echo '</ul>';
+    }
+
+    /**
+     * Display all the remaining plugins
+     */
+    protected function showPluginMenu() {
+        if(!count($this->menu)) return;
+        echo '<div class="clearer"></div>';
+        echo p_locale_xhtml('adminplugins');
+        echo '<ul class="admin_plugins">';
+        foreach ($this->menu as $item) {
+            $this->showMenuItem($item);
+        }
+        echo '</ul>';
+    }
+
+    /**
+     * Display the DokuWiki version
+     */
+    protected function showVersion() {
+        echo '<div id="admin__version">';
+        echo getVersion();
+        echo '</div>';
+    }
+
+    /**
+     * data security check
+     *
+     * simple check if the 'savedir' is relative and accessible when appended to DOKU_URL
+     *
+     * it verifies either:
+     *   'savedir' has been moved elsewhere, or
+     *   has protection to prevent the webserver serving files from it
+     */
+    protected function showSecurityCheck() {
+        global $conf;
+        if(substr($conf['savedir'], 0, 2) !== './') return;
+        echo '<a style="border:none; float:right;"
+                href="http://www.dokuwiki.org/security#web_access_security">
+                <img src="' . DOKU_URL . $conf['savedir'] . '/security.png" alt="Your data directory seems to be protected properly."
+                onerror="this.parentNode.style.display=\'none\'" /></a>';
+    }
+
+    /**
+     * Display a single Admin menu item
+     *
+     * @param array $item
+     */
+    protected function showMenuItem($item) {
+        global $ID;
+        if(blank($item['prompt'])) return;
+        echo '<li><div class="li">';
+        echo '<span>';
```

Comment:
Why is this in a span and not within the link? The icon is blue which indicates it's part of the link but it isn't. That is confusing. I would either make the colour not blue or put the icon inside the link to be clickable as well.